pkg/k8s/watchers/cilium_node.go

```diff
@@ -141,3 +145,37 @@ func (k *K8sWatcher) ciliumNodeInit(ciliumNPClient *k8s.K8sCiliumClient, asyncCo
 		log.Info("Disconnected from key-value store, restarting CiliumNode watcher")
 	}
 }
+
+// GetCiliumNode returns the CiliumNode "nodeName" from the local store. If the
+// local store is not initialized then it will fallback retrieving the node
+// from kube-apiserver.
+func (k *K8sWatcher) GetCiliumNode(ctx context.Context, nodeName string) (*cilium_v2.CiliumNode, error) {
+	var (
+		err                      error
+		nodeInterface            interface{}
+		exists, getFromAPIServer bool
+	)
+	k.ciliumNodeStoreMU.RLock()
+	if k.ciliumNodeStore == nil {
+		getFromAPIServer = true
+	} else {
+		nodeInterface, exists, err = k.ciliumNodeStore.GetByKey(nodeName)
+	}
+	k.ciliumNodeStoreMU.RUnlock()
+
+	if getFromAPIServer {
+		// fallback to using the kube-apiserver
+		return k8s.CiliumClient().CiliumV2().CiliumNodes().Get(ctx, nodeName, v1.GetOptions{})
+	}
+
+	if err != nil {
+		return nil, err
+	}
+	if !exists {
+		return nil, k8sErrors.NewNotFound(schema.GroupResource{
+			Group:    "cilium",
+			Resource: "CiliumNode",
+		}, nodeName)
+	}
+	return nodeInterface.(*cilium_v2.CiliumNode).DeepCopy(), nil
```

Comment:
Might be safer to type check and return and error otherwise.

Reply:
I don't think there's any way that `nodeInterface` is anything but `nil` or `*cilium_v2.CiliumNode` so I'd prefer a crash here as we cannot anyway continue operating if this fails. Long term I'd like to see all uses of shared stores wrapped with a typed variant that [expose only a safe API ](https://github.com/cilium/cilium/pull/21352/files#diff-3c74c464d90a898b07e840b7af98f573cb59222a587af5c77204a789ac684121).